src/DBFTables.jl

```diff
@@ -1,150 +1,293 @@
+
+    # the lengths are equal for each record
+    lengths = repeat(lengths_record, 1, header.records)
+    # the offsets accumulate over records with the record size
+    row_offsets = range(0; length = header.records, step = header.rsize)
+    offsets = repeat(offsets_record, 1, header.records)
+    offsets .+= reshape(row_offsets, 1, :)
+
+    StringArray{WeakRefString{UInt8},2}(data, offsets, lengths)
+end
+
+"Create a NamedTuple representing a single row"
+function Base.NamedTuple(row::Row)
+    dbf = gettable(row)
+    str = getstrings(dbf)
+    fields = getfields(dbf)
+    ncol = length(fields)
+    rowidx = getrow(row)
+    @inbounds record = @view str[:, rowidx]
+    @inbounds prs = (fields[col].name => dbf_value(
+        fields[col].type,
+        record[col],
+    ) for col = 1:ncol)
+    return (; prs...)
+end
+
+function Base.show(io::IO, row::Row)
+    show(io, NamedTuple(row))
+end
+
+function Base.show(io::IO, dbf::Table)
+    nr = length(dbf)
+    nc = length(getfields(dbf))
+    println(io, "DBFTables.Table with $nr rows and $nc columns")
+end
+
+Base.isempty(dbf::Table) = getheader(dbf).records == 0
+
+"Get a BitVector which is true for rows that are marked as deleted"
+function isdeleted(dbf::Table)
+    data = getfield(dbf, :data)
+    rsize = getheader(dbf).rsize
+    nrow = getheader(dbf).records
+    idx = range(1, step = rsize, length = nrow)
+    data[idx] .== 0x2a
+end
+
+"Check if the row is marked as deleted"
+function isdeleted(dbf::Table, row::Integer)
+    data = getfield(dbf, :data)
+    i = (row - 1) * getheader(dbf).rsize + 1
+    data[i] == 0x2a
+end
+
+"Iterate over the rows of a DBF Table, yielding a DBFTables.Row for each row"
+function Base.iterate(dbf::Table, st = 1)
+    st > length(dbf) && return nothing
+    return Row(dbf, st), st + 1
+end
+
+function Base.getproperty(row::Row, name::Symbol)
+    dbf = gettable(row)
+    str = getstrings(dbf)
+    colidx = findfirst(x -> x === name, propertynames(dbf))
+    colidx === nothing && throw(ArgumentError("Column not present: $name"))
+    type = @inbounds getfields(dbf)[colidx].type
+    rowidx = getrow(row)
+    return @inbounds dbf_value(type, str[colidx, rowidx])
+end
+
+Tables.istable(::Type{Table}) = true
+Tables.rowaccess(::Type{Table}) = true
+Tables.columnaccess(::Type{Table}) = true
+Tables.rows(dbf::Table) = dbf
+Tables.columns(dbf::Table) = dbf
+
+"Get the Tables.Schema of a DBF Table"
+function Tables.schema(dbf::Table)
+    names = Tuple(field.name for field in getfields(dbf))
+    # since missing is always supported, add it to the schema types
+    types = Tuple(Union{field.type,Missing} for field in getfields(dbf))
+    Tables.Schema(names, types)
+end
+
+"List all available DBF column names"
+Base.propertynames(dbf::Table) = getfield.(getfield(dbf, :header).fields, :name)
+Base.propertynames(row::Row) = propertynames(gettable(row))
+
+"Get an entire DBF column as a Vector. Usage: `dbf.myfield`"
```

Comment:
Probably worth mentioning in the doc here that each `dbf.myfield` makes a new copy of the column, so care should be taken to do field access a single time and subsequent operations should operate on the single reference.

Reply:
Yeah that's right, done in a2d719053a00080068cf8cf356967158836e67d2